File: services/ingestion-openmeteo/app/Application/Weather/Contracts/WeatherDataSource.php
<?php

namespace App\Application\Weather\Contracts;

use App\Application\Weather\DTO\WeatherSnapshotData;
use App\Domain\Weather\Location;

/**
 * Ovaj interfejs predstavlja apstrakciju nad izvorom vremenskih podataka.
 */
interface WeatherDataSource
{
    /**
     * Dohvata trenutni vremenski snapshot za zadatu lokaciju.
     */
    public function fetchCurrentWeather(Location $location): WeatherSnapshotData;
}

File: services/ingestion-weatherapi/internal/application/weather/contracts/weather_data_source.go
package contracts

import (
	"context"

	"ingestion-weatherapi/internal/domain/weather"
)

/*
WeatherDataSource predstavlja application contract za izvor vremenskih podataka.

Application sloj ne treba da zna da li podatke dobija iz WeatherAPI,
Open-Meteo, nekog trećeg HTTP API-ja ili mock implementacije.
*/
type WeatherDataSource interface {
	/*
		FetchCurrentWeather dohvaća trenutni vremenski snapshot za zadatu lokaciju.
	*/
	FetchCurrentWeather(ctx context.Context, location weather.Location) (weather.WeatherSnapshotData, error)
}
